### PR TITLE
[VL] Minor refactor for ColumnarBatchSerializer

### PR DESCRIPTION
**Pull request by Yohahaha**
## What changes were proposed in this pull request?

1. Rename ColumnarBatchSerializer to ColumnarShuffleSerializer, avoid use confused name with ColumnarBatchSerializerJniWrapper.
2. Rename ColumnarBatchSerializer to ColumnarBatchSerde, refactor API to separate serializer and deserializer.

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by Yohahaha** [comment]
@jinchengchenghh could you help take a look? thanks!

File: cpp/velox/operators/serializer/VeloxColumnarBatchSerde.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#include "operators/serializer/ColumnarBatchSerde.h"
+#include "velox/serializers/PrestoSerializer.h"
+
+namespace gluten {
+
+class VeloxColumnarBatchSerde final : public ColumnarBatchSerde {
+ public:
+  VeloxColumnarBatchSerde(std::shared_ptr<facebook::velox::memory::MemoryPool>, std::shared_ptr<arrow::MemoryPool>);
+
+  std::shared_ptr<ColumnarBatchSerializer> createSerializer() override;
```
Comment (marin-ma) [inline cpp/velox/operators/serializer/VeloxColumnarBatchSerde.h:32]: Why do we have `createSerializer` and `initDeserializer`?

File: cpp/velox/operators/serializer/VeloxColumnarBatchSerde.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  void initDeserializer(struct ArrowSchema* cSchema) override;
+
+  std::shared_ptr<ColumnarBatch> deserialize(uint8_t* data, int32_t size) override;
```
Comment (marin-ma) [inline cpp/velox/operators/serializer/VeloxColumnarBatchSerde.h:36]: I suppose there should be `ColumnarBatchDeserializer` for `createDeserializer`, and move `deserialize` there.

File: backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecHandler.scala [hunk 3]
```diff
@@ -260,7 +262,7 @@ class SparkPlanExecHandler extends SparkPlanExecApi {
         clazz.getConstructor(classOf[StructType], classOf[SQLMetric], classOf[SQLMetric])
       constructor.newInstance(schema, readBatchNumRows, numOutputRows).asInstanceOf[Serializer]
     } else {
-      new ColumnarBatchSerializer(schema, readBatchNumRows, numOutputRows, decompressTime)
+      new ColumnarShuffleSerializer(schema, readBatchNumRows, numOutputRows, decompressTime)
```
Comment (jinchengchenghh) [inline backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecHandler.scala:265]: Is this `ColumnarShuffleReader`? @zhztheplayer

**Pull request closed without merging**